nagios/nagios-events-wf.py

```diff
@@ -1,55 +1,100 @@
 
-def doMain():
-    msg = args.msg[0].replace("\\n", "\r").rstrip()
-    annotations = {'severity': args.type[0],
+
+def main():
+    msg = args.msg.replace("\\n", "\r").rstrip()
+    annotations = {'severity': args.type,
                    'type': 'Nagios',
                    'details': msg}
+    startTime = args.time
+    endTime = args.time + 1
 
-    notif = {'startTime': args.time[0],
-             'endTime': args.time[0] + 1,
-             'hosts': args.host,
-             'annotations': annotations}
-    if args.S is not None:
-        notif['name'] = "Nagios event on '{}' at '{}'".format(args.service, args.host[0])
+    if args.service is not None:
+        name = "Nagios event on '{}' at '{}'".format(args.service, args.host)
     else:
-        notif['name'] = "Nagios event at '{}'".format(args.host[0])
-
-    url = "{}/api/v2/event".format(args.server[0])
+        name = "Nagios event at '{}'".format(args.host)
 
-    headers = {'Content-Type': 'application/json',
-               'Content-Encoding': 'gzip',
-               'Authorization': 'Bearer ' + args.token[0]}
-    result = requests.post(url,
-                           headers=headers,
-                           data=json.dumps(notif))
-
-    if result.status_code != 200:
-        print(result.json())
-        sys.exit(-1)
-    else:
-        print("OK")
-        sys.exit(0)
+    wavefront_client.send_event(name, startTime, endTime, args.host, ["env:", "test-event"], annotations)
 
 
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    parser.add_argument('server', type=str, nargs=1, help='Wavefront Server URL.')
-    parser.add_argument('token', type=str, nargs=1, help='Wavefront token.')
+    parser = argparse.ArgumentParser(description='Sends Nagios perfdata to wavefront.')
+    parser.add_argument('--wf_server', dest="server", help='Wavefront Server URL.')
+    parser.add_argument('--wf_token', dest="token", help='Wavefront token.')
     parser.add_argument('-S', action='store_true', help='Service notification.')
-    parser.add_argument('--type', type=str, nargs=1, help='Notification type.')
-    parser.add_argument('--host', type=str, nargs=1, help='Notification hostname.')
+    parser.add_argument('--type', help='Notification type.')
+    parser.add_argument('--host', help='Notification hostname.')
     parser.add_argument('--service', type=str, nargs='?', help='Notification service name.')
-    parser.add_argument('--time', type=int, nargs=1, help='Notification time (UNIX epoch).')
-    parser.add_argument('--msg', type=str, nargs=1, help='Notification message.')
+    parser.add_argument('--time', type=int, help='Notification time (UNIX epoch).')
+    parser.add_argument('--msg', help='Notification message.')
+    parser.add_argument('--wf_proxy_addr', dest="addr")
+    parser.add_argument('--wf_proxy_port', dest="port")
+    parser.add_argument('--csp_base_url', dest="csp_base_url", help='Base URL of CSP console.')
```

Comment:
help='Base URL of the VMware Cloud Services Console.'

Isn't this static? Can we provide the actual URL?

Reply:
As the URL varies for staging and production, I did not hardcode the actual URL.
@gangadharaswamy Please share your thoughts